modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/item_actions/ItemActions.js

```diff
@@ -0,0 +1,249 @@
+ * details.
+ */
+
+import {openToast} from 'frontend-js-web';
+import React, {useContext, useState} from 'react';
+
+import DataSetContext from '../DataSetContext';
+import {ACTION_ITEM_TARGETS} from '../utils/actionItems/constants';
+import {formatActionURL} from '../utils/index';
+import {openPermissionsModal, resolveModalSize} from '../utils/modals/index';
+import ViewsContext from '../views/ViewsContext';
+import ActionsDropdown from './ActionsDropdown';
+import QuickActions from './QuickActions';
+import {itemActionsBasePropTypes} from './types/index';
+
+const {MODAL_PERMISSIONS} = ACTION_ITEM_TARGETS;
+
+const QUICK_ACTIONS_MAX_NUMBER = 3;
+
+export function isLink(target, onClick) {
+	return !(target && target !== 'link') && !onClick;
+}
+
+const formatActions = (actions, itemData) => {
+	return actions
+		? actions.reduce((actions, action) => {
+				if (action.data?.permissionKey) {
+					if (itemData.actions[action.data.permissionKey]) {
+						if (action.target === 'headless') {
+							return [
+								...actions,
+								{
+									...action,
+									...itemData.actions[
+										action.data.permissionKey
+									],
+								},
+							];
+						}
+						else {
+							return [...actions, action];
+						}
+					}
+
+					return actions;
+				}
+
+				return [...actions, action];
+		  }, [])
+		: [];
+};
+
+export function handleAction(
+	{
+		confirmationMessage,
+		event,
+		itemId,
+		method,
+		onClick,
+		setLoading,
+		size,
+		successMessage,
+		target,
+		title,
+		url,
+	},
+	{
+		executeAsyncItemAction,
+		highlightItems,
+		openModal,
+		openSidePanel,
+		toggleItemInlineEdit,
+	}
+) {
+	if (confirmationMessage && !confirm(confirmationMessage)) {
+		return;
+	}
+
+	if (target?.includes('modal')) {
+		event.preventDefault();
+
+		if (target === MODAL_PERMISSIONS) {
+			openPermissionsModal(url);
+		}
+		else {
+			openModal({
+				size: resolveModalSize(target),
+				title,
+				url,
+			});
+		}
+	}
+	else if (target === 'sidePanel') {
+		event.preventDefault();
+
+		highlightItems([itemId]);
+		openSidePanel({
+			size: size || 'lg',
+			title,
+			url,
+		});
+	}
+	else if (target === 'async' || target === 'headless') {
+		event.preventDefault();
+
+		setLoading(true);
+		executeAsyncItemAction(url, method)
+			.then(() => {
+				openToast({
+					message:
+						successMessage ||
+						Liferay.Language.get('action-completed'),
+					type: 'success',
+				});
+				setLoading(false);
+			})
+			.catch((_) => {
+				setLoading(false);
+			});
+	}
+	else if (target === 'inlineEdit') {
+		event.preventDefault();
+
+		toggleItemInlineEdit(itemId);
+	}
+	else if (target === 'blank') {
+		event.preventDefault();
+
+		window.open(url);
+	}
+	else if (onClick) {
+		event.preventDefault();
+
+		event.target.setAttribute('onClick', onClick);
+		event.target.onclick();
+		event.target.removeAttribute('onClick');
+	}
+}
+function ItemActions({actions, itemData, itemId}) {
```

Comment:
I still don't like that we have `actions` dir that has a completely different purpose. It's needlessly inconsistent. Maybe it would make sense to create new `/views/viewsReducer.js` and:
- put in it `viewsReducer` from `ViewsContext`
- merge to it everything from `/actions` dir
- move everything from `item_actions` into `actions`
- rename `ItemActions` to `Actions`, `ActionsMenu` or `ActionsPanel`